packages/core/contracts-ovm/insured-bridge/implementation/BridgeDepositBox.sol

```diff
@@ -182,22 +188,31 @@ abstract contract BridgeDepositBox is Legacy_Testable, Legacy_Lockable {
             getCurrentTime() >= quoteTimestamp - 10 minutes && getCurrentTime() <= quoteTimestamp + 10 minutes,
             "deposit mined after deadline"
         );
+        // If the address of the L1 token is the l1Weth and there is a msg.value with the transaction then the user
+        // is sending ETH. In this case, the ETH should be deposited to WETH, which is then bridged to L1.
+        if (whitelistedTokens[l2Token].l1Token == l1Weth && msg.value > 0) {
+            require(msg.value == amount, "msg.value must match amount");
+            l2Token.call{ value: msg.value }(abi.encodeWithSignature("deposit()"));
```

Comment:
Any reason not to just import the WETH abi and cast the address to it so this would be a normal function call?

Reply:
I had issues with doing it that way. will try again and report back with more detailed findings.